fix(template) Checkmate: Adds capture service to compose that allows localhost infrastructure monitoring

theodosiskats opened the pull request:
## Changes
- Added the official [Capture agent](https://github.com/bluewave-labs/capture) (`ghcr.io/bluewave-labs/capture:latest`) to the Checkmate template (`checkmate.yaml`).
- Exposes port `59232` and mounts `/etc/os-release` to allow system metrics collection.
- Configurable `API_SECRET` via environment variable (`CAPTURE_SECRET`).

## Issues
- The missing Capture agent prevented users from monitoring the localhost server infrastructure (CPU, memory, etc.).

### How to use
1. Define a strong secret for the agent in your environment (example with OpenSSL):
   openssl rand -hex 32

   Add it to your `.env` file:
   CAPTURE_SECRET=your-strong-random-secret

2. Deploy the updated template in Coolify.

3. In the Checkmate dashboard, go to Infrastructure → Create Monitor and set:
   - Server URL:
     http://capture:59232/api/v1/metrics
     (internal service name, works inside the Docker network)

   - Authorization secret:
     ${CAPTURE_SECRET}

4. Save the monitor.  
   Checkmate will now start showing host CPU, RAM, disk usage, and uptime metrics.
   
   <img width="1060" height="1132" alt="Screenshot 2025-08-25 at 1 31 09 AM" src="https://github.com/user-attachments/assets/c29cb259-61e0-414c-9373-34f935ecc438" />

theodosiskats committed:
Adds capture service to compose

Adds a new service named "capture" to the docker-compose file.

This service utilizes the `ghcr.io/bluewave-labs/capture:latest` image and exposes port 59232.
It also defines environment variables and a volume mount.

Cinzya commented on templates/compose/checkmate.yaml:
```diff
@@ -43,4 +43,14 @@ services:
+  capture:
+    image: ghcr.io/bluewave-labs/capture:latest
+    container_name: capture
+    restart: unless-stopped
```
remove those two. they are already applied by Coolify.

Cinzya commented on templates/compose/checkmate.yaml:
```diff
@@ -43,4 +43,14 @@ services:
+    container_name: capture
+    restart: unless-stopped
+    ports:
+      - "59232:59232"
```
is this port mapping necessary? It would be better if the we use the [magic envs](https://coolify.io/docs/knowledge-base/docker/compose#coolify-s-magic-environment-variables), specifically the FQDN.

theodosiskats committed:
Removes redundant container configuration

Removes the container_name, restart policy and port mapping
from the capture service definition. These configurations
are managed elsewhere, leading to duplication.

This change ensures a single source of truth for container
configurations, simplifying management and reducing potential
inconsistencies.

theodosiskats reviewed (commented):
as we use the docker internal service, port mapping is indeed redundant.

theodosiskats committed:
Merge branch 'next' into fix/missing-agent-from-checkmate-infra-monitor

theodosiskats committed:
Merge branch 'next' into fix/missing-agent-from-checkmate-infra-monitor

theodosiskats committed:
Merge branch 'next' into fix/missing-agent-from-checkmate-infra-monitor

theodosiskats committed:
Merge branch 'next' into fix/missing-agent-from-checkmate-infra-monitor

includenull commented:
Is it really worth adding to this template when the rest of it is outdated? `bluewaveuptime/uptime_client:latest` and `bluewaveuptime/uptime_redis:latest` are no longer used or being updated by the checkmate team.

Cinzya commented:
@includenull thanks for that callout. @theodosiskats do you think you could add this change in your PR as well? I believe [this](https://raw.githubusercontent.com/bluewave-labs/Checkmate/refs/heads/master/docker/dist/docker-compose.yaml) is their most recent example compose file, found it at least from their documentation. You can copy the correct images from there.

includenull commented:
They have an image called "mono" which is the frontend and backend in one container. I think that might work better for Coolify.

I currently have it running myself with docker compose and it's working well.

Cinzya commented:
> They have an image called "mono" which is the frontend and backend in one container. I think that might work better for Coolify.

Aah, good to know! I saw it mentioned on #6236 but wasn't sure if that is something additional. That might be better to use then indeed.

yipfram commented:
> 

What is your docker compose? Can you do a PR for it?

peaklabs-dev commented:
Thanks for the PR. I will close this as the template needs to be fully refactored.
